exe-common/src/sync.rs

```diff
@@ -273,13 +269,23 @@ fn net_sync_to<A: Api>(
                         tag::write(
                             &storage.read().unwrap(),
                             &tag::HEAD,
-                            &chain_state_before.last_block.as_ref(),
+                            &chain_state.last_block.as_ref(),
                         );
                     }
                 }
                 is_epoch_with_ebb = date.is_boundary()
             }
 
+            match chain_state.append_block(block_hash, block) {
```

Comment:
Also seems fine here since we never accessed chain_state until after this point anyway, and append_block() never generates the WrongPreviousBlock error triggering the other flow, plus chain_state_before will be identical before this line to chain_state so seems well.